libs/dal/src/repositories/message/message.schema.ts

```diff
@@ -121,6 +119,25 @@ const messageSchema = new Schema<MessageDBModel>(
   schemaOptions
 );
 
+/**
+ * todo: all the pre hooks should be removed after all the soft deletes are removed task nv-5688
+ */
+messageSchema.pre('find', function filterDeletedFind() {
+  this.where({ deleted: { $ne: true } });
+});
+messageSchema.pre('findOne', function filterDeletedFindOne() {
+  this.where({ deleted: { $ne: true } });
+});
+messageSchema.pre('findOneAndUpdate', function filterDeletedFindOneAndUpdate() {
+  this.where({ deleted: { $ne: true } });
+});
+messageSchema.pre('countDocuments', function filterDeletedCountDocuments() {
+  this.where({ deleted: { $ne: true } });
+});
+messageSchema.pre('count', function filterDeletedCount() {
+  this.where({ deleted: { $ne: true } });
+});
```

Comment:
Just to verify, that's the proper syntax? Just looks a bit weird. Shouldn't we return something here?

Reply:
it is i tested it locally, but we will still need to double test in staging

Reply:
So this just adds another where clause right? Not overrides the existing one

Reply:
correct